Additional docs for the OpenTelemetry Kotlin SDK (#9984)
open-telemetry/opentelemetry.io: content/en/docs/languages/kotlin/configuration.md

---

> title: Configure the SDK

Reviewer: ```suggestion
title: Configure OpenTelemetry Kotlin
```

linkTitle: Configure the SDK
weight: 13
---

## Configuring OpenTelemetry Kotlin

> The OpenTelemetry Kotlin SDK is configured at initialization through its DSL

Reviewer: Hugo renders the page title as specified in the front matter. To avoid the stacked headings, let's remove this one.

```suggestion
The OpenTelemetry Kotlin SDK is configured at initialization through its DSL
```

parameter. Both `createOpenTelemetry` and `createCompatOpenTelemetry` use the
same DSL, a full example of which is shown below:

```kotlin
val otel: OpenTelemetry = createOpenTelemetry {
    // configure SDK here
}
```

The following sections show how to configure different aspects of the SDK's
behavior. This is not an exhaustive list of ways to configure the SDK; rather it

> attempts to highlight the most common use-cases.

Reviewer: ```suggestion
attempts to highlight the most common use cases.
```

    }

    tracerProvider {
        resource {}
    }
}
```

It's also possible to scope a resource to a single signal. This is merged with
any global configuration (local takes priority):

```kotlin
val otel: OpenTelemetry = createOpenTelemetry {
    tracerProvider {
        resource {
            setStringAttribute("service.namespace", "payments")
        }
    }
}
```

Syntactic sugar also allows setting the Resources from a `Map<String, Any>`:

```kotlin
val otel: OpenTelemetry = createOpenTelemetry {
    resource {
        resource(mapOf("service.namespace" to "payments"))
    }
}
```

By default, the SDK always sets `service.name`, `service.version`, and
`telemetry.sdk.*`. These can be overriden by supplying your own value.

## Limits

### Attribute limits

[Attribute limits](/docs/specs/otel/common/#attribute-limits) limit the number
of attributes and the length of their values in characters. These can be
configured globally as shown below:

```kotlin
val otel: OpenTelemetry = createOpenTelemetry {
    attributeLimits {
        attributeCountLimit = 200
        attributeValueLengthLimit = 256
    }
}
```

### Log Limits

> [Log Limits](/docs/specs/otel/logs/sdk/#logrecord-limits) act as in the

Reviewer: ```suggestion
[Log Limits](/docs/specs/otel/logs/sdk/#logrecord-limits) act similarly to the
```
